src/Mod/Draft/draftutils/make_utils.py

```diff
@@ -0,0 +1,108 @@
+# ***************************************************************************
+# *   (c) 2019 Eliud Cabrera Castillo <e.cabrera-castillo@tum.de>           *
+# *                                                                         *
+# *   This file is part of the FreeCAD CAx development system.              *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   FreeCAD is distributed in the hope that it will be useful,            *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with FreeCAD; if not, write to the Free Software        *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+"""Provides utility functions for make_polarray and make_circulararray."""
+## @package make_utils
+# \ingroup DRAFT
+# \brief This module provides functions for make_polarray & make_circulararray.
+
+import FreeCAD as App
+import Part
+
+import draftutils.utils as utils
+
+from draftutils.messages import _msg, _err
+from draftutils.translate import _tr
+
+
+def make_polcirc_shared(_name, center, axis_object, axis_edge):
+    """Typecheck center, axis_object and axis_edge. Return axis_reference if
+    axis_object and axis_edge (optional) are set and correct. Otherwise return
+    None.
+
+    This is the shared functionality of make_polar_array and make_circulararray.
+    """
+    CORRECT = True
+    axis_reference = None
+
+    _msg("center: {}".format(center))
+    try:
+        utils.type_check([(center, App.Vector)], name=_name)
+    except TypeError:
+        _err(_tr("Wrong input: must be a vector."))
+        return not CORRECT, axis_reference
+
+    _msg("axis_object: {}".format(axis_object))
+    _msg("axis_edge: {}".format(axis_edge))
+    axis = axis_object
+    if axis_object and isinstance(axis_object, str):
+        found, axis = utils.find_object(axis_object,
+                                        doc=App.activeDocument())
```

Comment:
By the way, here you are using `App.activeDocument()`.

I think this should be in the upper function that calls this function.

See the beginning of `make_patharray`, the first call is a test for a document. Once we confirm this document exist, then we can pass the document everywhere else it is needed.
```py
    found, doc = utils.find_doc(App.activeDocument())
    if not found:
        _err(_tr("No active document. Aborting."))
        return None
    ...

    (all_correct,
     axis_reference) = make_utils.make_polcirc_shared(doc, _name,
                                                      center,
                                                      axis_object,
                                                      axis_edge)
```

Then inside the `make_polcirc_shared` we can use the same `doc` we already calculated outside; we don't need to call `App.activeDocument()` again.